src/library/parser.cpp

```diff
@@ -158,3 +170,50 @@ TrackFile Parser::playlistEntryToTrackFile(
         return TrackFile(QDir(basePath), filePath);
     }
 }
+
+bool Parser::exportPlaylistItemsIntoFile(
+        QString playlistFilePath,
+        const QList<QString>& playlistItemLocations,
+        bool useRelativePath) {
+    if (playlistFilePath.endsWith(
+                QStringLiteral(".pls"),
+                Qt::CaseInsensitive)) {
+        return ParserPls::writePLSFile(
+                playlistFilePath,
+                playlistItemLocations,
+                useRelativePath);
+    } else if (playlistFilePath.endsWith(
+                       QStringLiteral(".m3u8"),
+                       Qt::CaseInsensitive)) {
+        return ParserM3u::writeM3U8File(
+                playlistFilePath,
+                playlistItemLocations,
+                useRelativePath);
+    } else {
+        //default export to M3U if file extension is missing
+        if (!playlistFilePath.endsWith(
+                    QStringLiteral(".m3u"),
```

Comment:
The m3u playlist is Windows-1252 only.
Should we default to m3u8 to avoid data loss? 

The attempt to export a m3u playlist with invalid charters fails with a message box.

Reply:
off-topic but related: I think we should have a helper class to check for valid filenames that alows to reduce redundancy (mapping export, playlist export, effect chain export and what not).

Reply:
I already added helper functions here to ensure a generated filename is readable on all platforms (afaik), as well as template generators and escape filters.
i would like to later consolidate the crates export (djinterop) to just be part of this export. later also transcoding into different fileformats while exporting (cdj compatible format (aiff/wav)), etc.